presences/src/main/java/fr/openent/presences/service/impl/DefaultAlertService.java

```diff
@@ -134,9 +134,17 @@ public Future<JsonArray> getAlertsStudents(String structureId, List<String> type
             params.add(endTime);
         }
 
-        query += " GROUP BY student_id, type HAVING count(*) >= " + Presences.dbSchema + ".get_alert_thresholder(type, ?);";
+        query += " GROUP BY student_id, type HAVING count(*) >= " + Presences.dbSchema + ".get_alert_thresholder(type, ?)";
         params.add(structureId);
 
+        if(page != null){
+            query += " OFFSET ? LIMIT ?;";
+            params.add(Presences.PAGE_SIZE * page);
+            params.add(Presences.PAGE_SIZE);
+        }else{
+            query += ";";
+        }
```

Comment:
```suggestion
        if(page != null){
            query += " OFFSET ? LIMIT ?";
            params.add(Presences.PAGE_SIZE * page);
            params.add(Presences.PAGE_SIZE);
        }
        
        query = ";"
```
ça t'évite un else.
Par la même occase, la pagination en SQL étant un sujet régulier, tu devrais être en mesure d'externaliser ça peut-être (dans un helper par exemple)